File: 6-Tarea-Arreglos-matrices/D.Conjunto-Capicua.cpp
#include <iostream>
using namespace std;

int main() {
  int N;
  cin >> N;

  int arr[N];
  for (int i = 0; i < N; i++)
    cin >> arr[i];

  bool capicua = true;
  for (int i = 0; i < N; i++)
    capicua = capicua && (arr[i] == arr[N - 1 - i]);

  cout << (capicua ? "SI" : "NO") << endl;

  return 0;
}

File: 6-Tarea-Arreglos-matrices/D.Conjunto-Capicua.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);

    int N = scanner.nextInt();
    int[] arr = new int[N];
    
    for (int i = 0; i < N; i++)
      arr[i] = scanner.nextInt();

    boolean capicua = true;
    for (int i = 0; i < N; i++)
      capicua = capicua && (arr[i] == arr[N - 1 - i]);

    System.out.println((capicua ? "SI" : "NO"));
  }
}